crawler.py

```diff
@@ -189,94 +189,102 @@ def run(website, driver):
+                    # Zip gcda file from gcov directory
+                    shutil.make_archive('code-coverage-gcda', 'zip', gcov_dir)
+                    grcov_command = [
+                        os.path.join('tools', 'grcov'),
+                        '-t', 'coveralls+',
+                        '-p', prefix,
+                        os.path.join('tools', 'target.code-coverage-gcno.zip'), 'code-coverage-gcda.zip',
+                        jsvm_output_file,
+                        '--filter', 'covered',
+                        '--token', 'UNUSED',
+                        '--commit-sha', 'UNUSED'
+                    ]
 
-    # All steps are stored in new folder
-    data_folder = str(uuid.uuid4())
-    run_all(driver, data_folder)
+                    with open('output.json', 'w+') as outfile:
+                        subprocess.check_call(grcov_command, stdout=outfile)
 
-    sys.path.insert(0, 'tools/mozbuild/codecoverage')
+                    with open('tests_report.json') as baseline_rep, open('output.json') as rep:
+                        baseline_report = json.load(baseline_rep)
+                        report = json.load(rep)
 
-    from lcov_rewriter import LcovFileRewriter
+                    filterpaths.ignore_third_party_filter(report)
 
-    jsvm_output_file = 'jsvm_lcov_output.info'
-    jsvm_files = [os.path.join(jsvm_dir, e) for e in os.listdir(jsvm_dir)]
-    rewriter = LcovFileRewriter(os.path.join('tools', 'chrome-map.json'))
-    rewriter.rewrite_files(jsvm_files, jsvm_output_file, '')
+                    # Create diff report
+                    diff_report = diff.compare_reports(baseline_report, report, True)
 
-    # Zip gcda file from gcov directory
-    shutil.make_archive('code-coverage-gcda', 'zip', gcov_dir)
-    grcov_command = [
-        os.path.join('tools', 'grcov'),
-        '-t', 'coveralls+',
-        '-p', prefix,
-        os.path.join('tools', 'target.code-coverage-gcno.zip'), 'code-coverage-gcda.zip',
-        jsvm_output_file,
-        '--filter', 'covered',
-        '--token', 'UNUSED',
-        '--commit-sha', 'UNUSED'
-    ]
+                    with open('{}/diff.json'.format(data_folder), 'w') as outfile:
+                        json.dump(diff_report, outfile)
 
-    with open('output.json', 'w+') as outfile:
-        subprocess.check_call(grcov_command, stdout=outfile)
+                    for filename in ['code-coverage-gcda.zip', jsvm_output_file]:
+                        os.remove(filename)
 
-    with open('tests_report.json') as baseline_rep, open('output.json') as rep:
-        baseline_report = json.load(baseline_rep)
-        report = json.load(rep)
+                    generatehtml.generate_html(data_folder)
 
-    filterpaths.ignore_third_party_filter(report)
+                    # Clean directories
+                    [shutil.rmtree(os.path.join(gcov_dir, f)) for f in os.listdir(gcov_dir)]
+                    [os.remove(os.path.join(jsvm_dir, f)) for f in os.listdir(jsvm_dir)]
```

Comment:
You can simply do:
```Python
shutil.rmtree(gcov_dir)
shutil.rmtree(jsvm_dir)
```
instead.

Reply:
This will remove folders too but I'm reusing them for the next loop iteration.

Reply:
Actually, you could move the temporary dir creation in the for loop, this way you don't have to manually clean the directories at the end.